thingsvision/core/extraction/torch.py

```diff
@@ -230,3 +232,25 @@ def get_default_transformation(
 
     def get_backend(self) -> str:
         return "pt"
+
+
+class BatchExtraction(object):
+
+    def __init__(
+        self, extractor: PyTorchExtractor, module_name: str, output_type: str
+    ) -> None:
+        self.extractor = extractor
+        self.module_name = module_name
+        self.output_type = output_type
+
+    def __enter__(self) -> PyTorchExtractor:
+        self.extractor._module_and_output_check(self.module_name, self.output_type)
+        self.extractor.register_hook(self.module_name)
```

Comment:
Minor suggestion: `self.extractor.register_hook` and `self.extractor._unregister_hook()` either both private or public object methods.

Reply:
Both are private methods now.